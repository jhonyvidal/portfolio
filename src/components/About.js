import React from 'react';
import CountUp from 'react-countup';
import { useInView } from 'react-intersection-observer'
import {motion} from 'framer-motion';
import { fadeIn } from '../variants';
import { useNavigate } from "react-router-dom";
import '../i18n/config';
import { useTranslation } from 'react-i18next';

const About = () => {
  const { t } = useTranslation();
  const [ref,inView] = useInView({
    threshould:0.5,
  })
  const navigate = useNavigate();
  const redirect = () => {
    navigate('/educations', { replace: true })
  };
  const redirectToSkills = (event) => {
    event.preventDefault();
    navigate('/skills', { replace: true })
  };
  return (
  <section className='section' id="about" ref={ref}>
    <div className='container mx-auto'>
      <div className='flex flex-col gap-y-10 lg:flex-row lg:items-center lg:gap-x-20 lg:gap-y-0 h-screen'>
        {/* img */}
        <motion.div
        variants={fadeIn('right',0.3)}
        initial="hidden"
        whileInView={'show'}
        viewport={{once:false,amount:0.3}} 
        className='flex-1 bg-about bg-contain bg-no-repeat h-[640px] mix-blend-lighten bg-top'>
          
        </motion.div>
        {/* text */}
        <motion.div 
        variants={fadeIn('left',0.3)}
        initial="hidden"
        whileInView={'show'}
        viewport={{once:false,amount:0.3}} 
        className='flex-1'>
          
          <h2 className='h2 text-accent'>{t('about.title')}</h2>
          <h3 className='h3 mb-4'>{t('about.subtitle')}</h3>
          <p className='mb-6'>{t('about.text')} </p>
          {/* stats */}
          <div className='flex gap-x-6 lg:gap-x-10 mb-12'>
            <div>
              <div className='text-[40px] font-tertiary text-gradient mb-2'>
                {
                  inView ? <CountUp start={0}  end={6} duration={3}/>
                  :null
                }
              </div>
              <div className='font-primary text-sm tracking-[2px]'> 
                <p dangerouslySetInnerHTML={{ __html: t('about.item1') }} />
              </div>
            </div>
            <div>
              <div className='text-[40px] font-tertiary text-gradient mb-2'>
                {
                  inView ? <CountUp start={0}  end={20} duration={3}/>
                  :null
                }+
              </div>
              <div className='font-primary text-sm tracking-[2px]'> 
                <p dangerouslySetInnerHTML={{ __html: t('about.item2') }} />
              </div>
            </div>
            <div>
              <div className='text-[40px] font-tertiary text-gradient mb-2'>
                {
                  inView ? <CountUp start={0}  end={50} duration={3}/>
                  :null
                }+
              </div>
              <div className='font-primary text-sm tracking-[2px]'> 
                <p dangerouslySetInnerHTML={{ __html: t('about.item3') }} />
              </div>
            </div>
          </div>
          <div className='flex gap-x-8 items-center'>
            <button className='btn btn-lg ' onClick={redirect}>{t('myeducations')}</button>
            <a onClick={redirectToSkills} className='text-gradient btn-link'  href='#'>
            {t('myskills')} 
            </a>
          </div>
        </motion.div>    
      </div>
    </div>
  </section>
  );
};

export default About;
