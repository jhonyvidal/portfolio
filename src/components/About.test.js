import About from './About';
import { render,fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';

// Mocking react-router-dom useNavigate
jest.mock('react-router-dom', () => ({
    useNavigate: () => jest.fn()
}));

test('renders About component', () => {
    const {queryByLabelText, getByLabelText} = render(
        <About labelOn="On" labelOff="Off" />,
    );
    expect(queryByLabelText(/off/i)).toBeTruthy();

    fireEvent.click(getByLabelText(/off/i));
  
    expect(queryByLabelText(/on/i)).toBeTruthy();
 
});

// test('redirects to /educations when "My Educations" button is clicked', () => {
//   const navigate = jest.fn();
//   const { getByText } = renderer.create(<About />);
//   const educationsButton = getByText(/myeducations/i);
//   fireEvent.click(educationsButton);
//   expect(navigate).toHaveBeenCalledWith('/educations', { replace: true });
// });