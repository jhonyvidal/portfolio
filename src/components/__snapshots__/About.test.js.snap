// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders About component 1`] = `
<section
  className="section"
  id="about"
>
  <div
    className="container mx-auto"
  >
    <div
      className="flex flex-col gap-y-10 lg:flex-row lg:items-center lg:gap-x-20 lg:gap-y-0 h-screen"
    >
      <div
        className="flex-1 bg-about bg-contain bg-no-repeat h-[640px] mix-blend-lighten bg-top"
        style={
          Object {
            "opacity": 0,
            "transform": "translateX(-80px) translateY(0px) translateZ(0)",
          }
        }
      />
      <div
        className="flex-1"
        style={
          Object {
            "opacity": 0,
            "transform": "translateX(80px) translateY(0px) translateZ(0)",
          }
        }
      >
        <h2
          className="h2 text-accent"
        >
          About me.
        </h2>
        <h3
          className="h3 mb-4"
        >
          i'm a Fullstack Developer with over 6 years of experience
        </h3>
        <p
          className="mb-6"
        >
          I have created and developed differents applications with long techonologies such as React, Angular, Ionic, React Native, C#, .Net, PHP, Java and Python
           
        </p>
        <div
          className="flex gap-x-6 lg:gap-x-10 mb-12"
        >
          <div>
            <div
              className="text-[40px] font-tertiary text-gradient mb-2"
            />
            <div
              className="font-primary text-sm tracking-[2px]"
            >
              <p
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Years of<br/>Experience",
                  }
                }
              />
            </div>
          </div>
          <div>
            <div
              className="text-[40px] font-tertiary text-gradient mb-2"
            >
              +
            </div>
            <div
              className="font-primary text-sm tracking-[2px]"
            >
              <p
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Projects<br/>Completed",
                  }
                }
              />
            </div>
          </div>
          <div>
            <div
              className="text-[40px] font-tertiary text-gradient mb-2"
            >
              +
            </div>
            <div
              className="font-primary text-sm tracking-[2px]"
            >
              <p
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "Projects<br/>Completed",
                  }
                }
              />
            </div>
          </div>
        </div>
        <div
          className="flex gap-x-8 items-center"
        >
          <button
            className="btn btn-lg "
            onClick={[Function]}
          >
            My Educations
          </button>
          <a
            className="text-gradient btn-link"
            onClick={[Function]}
          >
            My Skills
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;
